src/nvofapi/nvofapi_common.cpp

```diff
@@ -0,0 +1,218 @@
+                initParams->inputBufferFormat);
+
+        VkOpticalFlowSessionCreateInfoNV createInfo{};
+        createInfo.sType = VK_STRUCTURE_TYPE_OPTICAL_FLOW_SESSION_CREATE_INFO_NV;
+        createInfo.width = initParams->width;
+        createInfo.height = initParams->height;
+        createInfo.outputGridSize = VK_OPTICAL_FLOW_GRID_SIZE_4X4_BIT_NV;
+
+        switch (initParams->perfLevel) {
+            case NV_OF_PERF_LEVEL_SLOW:
+                createInfo.performanceLevel = VK_OPTICAL_FLOW_PERFORMANCE_LEVEL_SLOW_NV;
+                break;
+            case NV_OF_PERF_LEVEL_MEDIUM:
+                createInfo.performanceLevel = VK_OPTICAL_FLOW_PERFORMANCE_LEVEL_MEDIUM_NV;
+                break;
+            case NV_OF_PERF_LEVEL_FAST:
+                createInfo.performanceLevel = VK_OPTICAL_FLOW_PERFORMANCE_LEVEL_FAST_NV;
+                break;
+            default:
+                createInfo.performanceLevel = VK_OPTICAL_FLOW_PERFORMANCE_LEVEL_UNKNOWN_NV;
+                break;
+        }
+
+        switch (initParams->inputBufferFormat) {
+            case NV_OF_BUFFER_FORMAT_GRAYSCALE8:
+                createInfo.imageFormat = VK_FORMAT_R8_UNORM;
+                break;
+            case NV_OF_BUFFER_FORMAT_NV12:
+                createInfo.imageFormat = VK_FORMAT_G8_B8R8_2PLANE_420_UNORM;
+                break;
+            case NV_OF_BUFFER_FORMAT_ABGR8:
+                createInfo.imageFormat = VK_FORMAT_B8G8R8A8_UNORM;
+                break;
+            default:
+                createInfo.imageFormat = VK_FORMAT_UNDEFINED;
+                break;
+        }
+
+        // Need to get the size/id for the private data to pass it along to VK...
+        VkOpticalFlowSessionCreatePrivateDataInfoNV privData{};
+        privData.sType = VK_STRUCTURE_TYPE_OPTICAL_FLOW_SESSION_CREATE_PRIVATE_DATA_INFO_NV;
+        privData.size = ((NV_OF_PRIV_DATA*)initParams->hPrivData)->size;
+        privData.id = ((NV_OF_PRIV_DATA*)initParams->hPrivData)->id;
+        privData.pPrivateData = ((NV_OF_PRIV_DATA*)initParams->hPrivData)->data;
+
+        createInfo.pNext = &privData;
+
+        auto ret = m_vkCreateOpticalFlowSessionNV(m_vkDevice, &createInfo, NULL, &m_vkOfaSession);
+
+        if (ret == VK_SUCCESS) {
+            return Success();
+        }
+
+        return ErrorGeneric();
+    }
+
+    NV_OF_STATUS NvOFInstance::BindImageToSession(NvOFImage* image, VkOpticalFlowSessionBindingPointNV bindingPoint) {
+        VkImageLayout layout = VK_IMAGE_LAYOUT_GENERAL;
+
+        if (!image)
+            return ErrorGeneric();
+
+        auto ret = m_vkBindOpticalFlowSessionImageNV(m_vkDevice,
+            m_vkOfaSession,
+            bindingPoint,
+            image->ImageView(),
+            layout);
+        if (ret != VK_SUCCESS) {
+            return ErrorGeneric();
+        }
+        return Success();
+    }
+
+    NV_OF_STATUS NvOFInstance::getCaps(NV_OF_CAPS param, uint32_t* capsVal, uint32_t* size) {
```

Comment:
I suppose it technically could right now, but eventually a full implementation of it would make use of `m_vkPhysicalDevice` to query information about the optical flow capabilities of the desired GPU. I'm not sure that adding the static keyword now provides much value, although neither does the current (mostly-stub) implementation.